router/router.go

```diff
@@ -672,16 +672,29 @@ func (worker *workerT) processDestinationJobs() {
 							} else {
 								// stat start
 								pkgLogger.Debugf(`responseTransform status :%v, %s`, worker.rt.transformerProxy, worker.rt.destName)
-								sendCtx, cancel := context.WithTimeout(ctx, worker.rt.netClientTimeout)
-								defer cancel()
+								var sendCtx context.Context
+								var cancel context.CancelFunc
 								// transformer proxy start
 								if worker.rt.transformerProxy {
-									jobID := destinationJob.JobMetadataArray[0].JobID
-									pkgLogger.Debugf(`[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Request started`, worker.rt.destName, jobID)
+									jobId := destinationJob.JobMetadataArray[0].JobID
+									pkgLogger.Debugf(`[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Request started`, worker.rt.destName, jobId)
+									proxyReqparams := &transformer.ProxyRequestParams{
+										DestName:     worker.rt.destName,
+										JobId:        jobId,
+										ResponseData: val,
+									}
+									// For http.client for proxy we are using
+									// backendProxyTimeout + netClientTimeout as the timeout value
+									// we are adding 20ms extra here to make sure the context timeout is at higher value
+									// Should we reduce the value from 20 to 10 ms ?
+									prxCtxTimeout := worker.rt.backendProxyTimeout + worker.rt.netClientTimeout + 20*time.Millisecond
+									sendCtx, cancel = context.WithTimeout(ctx, prxCtxTimeout)
+									defer cancel()
```

Comment:
usage of `defer` here is incorrect, we don't want `cancel` to be called after `processDestinationJobs` is done, we want it to be called after the http request code block finishes.
```suggestion
```

Reply:
Done